python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -968,10 +967,16 @@ def has_loaded(self, obj):
         Args:
             obj (GangaObject): Object which we want to look for in this repo
         """
+        obj_id = id(obj)
+        self.lock_transaction(obj_id, 'has_loaded')
```

Comment:
I might be misunderstanding the `lock_transaction` method but doesn't this only protect against altering the registry if the same object is being operated on? e.g. it would stop you removing the object that was being checked for with `has_loaded`. Isn't the bigger problem that the `_objects` dictionary changes while the underlying `find` is going on? e.g. you call `has_loaded` with one object but are removing another at the same time. That's still allowed by this transaction lock but would be a problem overall. I would expect (though this may be wrong) if you call `remove` on an object, that requires a lock on the object itself which would stop it's changing from outside anyway.